nx/lib/nx.ex

```diff
@@ -3985,6 +4000,121 @@ defmodule Nx do
 
   def devectorize(tensor), do: tensor
 
+  @doc """
+  Reshapes input tensors so that they are all vectorized with the same vectors.
```

Comment:
I think the most important here is the order. `reshape_vectors([x, y])` is different from `[y, x]` in the example below. so I would even change the second example to be `[y, x]`instead of the more complex example we have today?

Reply:
Done